### PR TITLE
Update relative-urls.php

### PR DESCRIPTION
**Pull request by z-avanes**
some sitemap plugin like yoast seo generate url for sitemap like :
http://domain.com/post-sitemap1.xml
http://domain.com/page-sitemap.xml
http://domain.com/post_tag-sitemap4.xml
...
and prev regex cannot handel this type of url and cause of error in google webmaster
the new regex fix this problem

### CODE REVIEW

File: modules/relative-urls.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * add_theme_support('soil-relative-urls');
  */
  
-if (is_admin() || preg_match('/sitemap(_index)?\.xml/', $_SERVER['REQUEST_URI']) || in_array($GLOBALS['pagenow'], ['wp-login.php', 'wp-register.php'])) {
+if (is_admin() || preg_match('/([a-zA-Z0-9_-]*)sitemap(_index)?(\d*)\.xml/', $_SERVER['REQUEST_URI']) || in_array($GLOBALS['pagenow'], ['wp-login.php', 'wp-register.php'])) {
```
Comment (swalkinshaw) [inline modules/relative-urls.php]: Not a huge deal but this regexp could be a little more efficient by losing the capture groups which aren't used:

`/^\/[a-zA-Z0-9_-]*sitemap(_index)?\d*\.xml$/`

Comment (QWp6t) [inline modules/relative-urls.php]: We don't need the first part either. This should work.

`sitemap(_index)?\d*\.xml$`

https://regex101.com/r/sI7sF6/2

Comment (z-avanes) [inline modules/relative-urls.php]: i think @swalkinshaw offer is good
/^\/[a-zA-Z0-9_-]*sitemap(_index)?\d*\.xml$/

Comment (retlehs) [inline modules/relative-urls.php]: could you please update this PR with that regex and squash the commits? thanks!

Comment (z-avanes) [inline modules/relative-urls.php]: regex updated ;-)

Comment (retlehs) [inline modules/relative-urls.php]: thanks, could you please squash your commits? 

https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

Comment (z-avanes) [inline modules/relative-urls.php]: thanks for help 
i am beginner in git and make mistake with this PR so close and create new PR

**Pull request closed without merging**